core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -51,11 +88,8 @@ private[spark] object Signaling extends Logging {
       // register old handler, will receive incoming signals while this handler is running
       Signal.handle(signal, prevHandler)
 
-      val escalate = actions.asScala forall { action =>
-        !action()
-      }
-
-      if(escalate) {
+      val escalate = actions.asScala.forall { action => !action() }
```

Comment:
can you write a short description here on what this is doing? it's not very straightforward (especially the forall and boolean inversion)